### PR TITLE
port Pad to prototype transforms

### LINKED ISSUE #5521: Port `transforms.Pad` to `prototype.transforms`
cc @vfdev-5 @datumbox @bjuncek

### PR DESCRIPTION
**Pull request by pmeier**
Closes #5521.

### CODE REVIEW

File: torchvision/prototype/transforms/_geometry.py [hunk 1]
```diff
@@ -272,42 +273,31 @@ def apply_recursively(obj: Any) -> Any:
...
 
 
-class RandomZoomOut(Transform):
+class Pad(Transform):
     def __init__(
-        self, fill: Union[float, Sequence[float]] = 0.0, side_range: Tuple[float, float] = (1.0, 4.0), p: float = 0.5
+        self,
+        padding: Union[int, Sequence[int]],
+        fill: Union[float, Sequence[float]] = 0.0,
+        padding_mode: str = "constant",
```
Comment (datumbox) [inline torchvision/prototype/transforms/_geometry.py:281]: Nit: What happened to `Literal`? :)

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

File: torchvision/prototype/transforms/_geometry.py (hunk 1 above)
Comment (pmeier) [reply torchvision/prototype/transforms/_geometry.py:281]: I forgot about it :slightly_smiling_face:

File: torchvision/prototype/transforms/_geometry.py [hunk 2]
```diff
@@ -272,42 +273,31 @@ def apply_recursively(obj: Any) -> Any:
...
-        self.p = p
-
-    def _get_params(self, sample: Any) -> Dict[str, Any]:
-        image = query_image(sample)
-        orig_c, orig_h, orig_w = get_image_dimensions(image)
-
-        r = self.side_range[0] + torch.rand(1) * (self.side_range[1] - self.side_range[0])
-        canvas_width = int(orig_w * r)
-        canvas_height = int(orig_h * r)
+        if not isinstance(fill, (numbers.Number, str, tuple, list)):
```
[the remark this answers is not in the document]
Comment (pmeier) [reply torchvision/prototype/transforms/_geometry.py:287]: What kind of str were supported here? TBH, I mostly copy-pasted form teh old transform and missed that part.

[the remark this answers is not in the document]
Comment (datumbox) [reply torchvision/prototype/transforms/_geometry.py:287]: PIL supports colours like `"FF00FF"`. Perhaps the way to resolve this comment is add an entry on the clean up issue that you maintain to put deprecation warnings for all the colours provided in this format.

[the remark this answers is not in the document]
Comment (pmeier) [reply torchvision/prototype/transforms/_geometry.py:288]: See #5627.

File: torchvision/prototype/transforms/_geometry.py [hunk 3]
```diff
@@ -272,42 +274,31 @@ def apply_recursively(obj: Any) -> Any:
         return apply_recursively(inputs if len(inputs) > 1 else inputs[0])
 
 
-class RandomZoomOut(Transform):
+class Pad(Transform):
```
[the remark this answers is not in the document]
Comment (pmeier) [reply torchvision/prototype/transforms/_geometry.py:277]: Yep, this is broken. We never noticed, because the tests are broken too. I'll have a look.

**Pull request merged**